### PR TITLE
backend: create delete assessment resolver

### PR DESCRIPTION
**Pull request by brian386**
## Notion ticket link

[Create resolver to Delete Assessment](https://www.notion.so/uwblueprintexecs/Create-resolver-to-Delete-Assessment-7689441452d345aa84cc89f776f579b9)

## Implementation description
* 

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/models/test.model.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ export enum QuestionComponentType {
 }
 
 export enum AssessmentType {
-  BEGINNING,
-  END,
+  BEGINNING = "BEGINNING",
```
Comment (brian386) [inline backend/models/test.model.ts:16]: Fixing mistake from previous PR. Typescript enums are integers but graphql enums are strings so I was getting invalid enum value error.

File: backend/graphql/resolvers/testResolvers.ts [hunk 2]
```diff
@@ -148,6 +148,21 @@ const testResolvers = {
     ): Promise<TestResponseDTO | null> => {
       return testService.duplicateTest(id);
     },
+    deleteAssessment: async (
+      _req: undefined,
+      { id }: { id: string },
+    ): Promise<string> => {
+      const testToDelete = await testService.getTestById(id);
```
Comment (carissa-tang) [inline backend/graphql/resolvers/testResolvers.ts]: sorry for not mentioning this earlier... but do you mind moving this logic into the service directly? (i.e. updating `testService.deleteTest(id)` to do either a soft or hard delete depending on status)

Comment (carissa-tang) [reply backend/graphql/resolvers/testResolvers.ts]: this way, you will not need to write a resolver (since `deleteTestById` already exists)

File: graphqlTestExample.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+updateTest(id:"63fc1b3774319e02fbe49775" test:{
```
Comment (carissa-tang) [inline graphqlTestExample.txt]: what is this for?

Comment (carissa-tang) [reply graphqlTestExample.txt]: if you did some testing, feel free to just add a screenshot to your PR / attach code there! you don't need to commit it in the code

File: backend/models/test.model.ts [hunk 4]
```diff
@@ -137,12 +138,12 @@ const TestSchema: Schema = new Schema(
     assessmentType: {
       type: String,
       required: true,
-      enum: Object.keys(AssessmentType),
```
Comment (carissa-tang) [inline backend/models/test.model.ts]: is this change necessary? i believe it should work with the enum

File: backend/testUtils/tests.ts [hunk 5]
```diff
@@ -136,10 +136,10 @@ export const assertResponseMatchesExpected = (
 ): void => {
   expect(result.id).not.toBeNull();
   expect(result.name).toEqual(expected.name);
-  expect(result.assessmentType).toEqual(expected.assessmentType.toString());
```
Comment (carissa-tang) [inline backend/testUtils/tests.ts:139]: is this change necessary?

File: graphqlTestExample.txt (hunk 3 above)
Comment (brian386) [reply graphqlTestExample.txt]: oops uh I meant to delete this after but I forgot

File: backend/models/test.model.ts (hunk 4 above)
Comment (brian386) [reply backend/models/test.model.ts]: no it's not, I was just trying things - I can change it back

File: backend/testUtils/tests.ts (hunk 5 above)
Comment (brian386) [reply backend/testUtils/tests.ts:139]: I added toString() in my previous PR because the enum used to be an integer. But now I changed the enum to be a string so I thought it's weird to toString() a string.

File: backend/services/implementations/__tests__/testService.test.ts [hunk 6]
```diff
@@ -155,7 +155,7 @@ describe("mongo testService", (): void => {
 
     const originalTest = await MgTest.findById(test.id);
     // TODO: update this to be soft delete instead of hard delete
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/testService.test.ts:157]:
```suggestion
```

File: backend/testUtils/tests.ts (hunk 5 above)
Comment (carissa-tang) [reply backend/testUtils/tests.ts:139]: ahh makes sense!

**Review by carissa-tang — APPROVED** [review]
just a small change, but lgtm!
